### ISSUE TITLE
Potential accuracy issue native_batch_norm decomp

### ISSUE THREAD

**Issue by anijain2305** [pytorch/pytorch#93433]
### 🐛 Describe the bug

The following two cmds fail accuracy

`python benchmarks/dynamo/timm_models.py  --accuracy --device cuda --backend=aot_inductor_debug --training --float32 --only=resnest101e`

`python benchmarks/dynamo/torchbench.py --training --accuracy --device cuda --backend=aot_inductor_debug --amp --only=mobilenet_v3_large` (fixed by https://github.com/pytorch/pytorch/issues/88375)

`python benchmarks/dynamo/torchbench.py --training --accuracy --device cuda --backend=aot_inductor_debug --float32 --only=resnet50_quantized_qat`

And both of them pass if I disable the `aten.native_batch_norm` in torch/_inductor/decomposition.py

Note that the backend here is `aot_inductor_debug` which only uses the decomp and partitioner from inductor but does not use inductor codegen.

Minifiers are unable to help here.

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh
